lib/handler/pool/state.ex

```diff
@@ -15,7 +15,9 @@ defmodule Handler.Pool.State do
           workers: %{
             reference() => {
               bytes_committed :: non_neg_integer(),
-              pid()
+              from_pid :: pid(),
+              task_pid :: pid(),
+              task_unique_name :: String.t()
```

Comment:
Hmmmm good questions 🤔 

> Would we never return `{:reject, _}`

I don't think we ever would. If there are no matching tasks we would return `{:ok, 0}` meaning nothing was killed, but no error occurred. 

> every pool that could possible have that task would need to kill it

I'm assuming that when we kill a job, we know which customer that job was for. In our current use-case the pools look like this:

<img width="365" alt="Screenshot 2022-01-28 at 13 13 33" src="https://user-images.githubusercontent.com/80008/151552949-6e79af90-3ecc-4bd2-9881-742d8e130a5c.png">

So in that case we would need to send a `kill` message to the customer-specific pool and for any matching jobs we would also need to send a `kill` message to the shared pool. But we wouldn't need to send any messages to the other 3 customer pools.

In the future when we can delegate to shared pools on multiple different servers, the scenario would look more like this:

<img width="465" alt="Screenshot 2022-01-28 at 13 15 40" src="https://user-images.githubusercontent.com/80008/151553223-df5fcb6d-a3f4-459c-8081-fa1d8be9c9f5.png">

But in that case I think each of the tasks in the customer-pool will need to keep track of which upstream pools each task is running in, so it would mean that we have to send a kill message once to the customer pool, and it may need to forward that to a few of the shared pools (as many as are actually handling those tasks)

Does that all sound right to you?